components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/response/SAMLLogoutResponseFactory.java

```diff
@@ -0,0 +1,97 @@
+
+        return (exception instanceof SAMLIdentityException) && ((SAMLIdentityException) exception).getAcsUrl() != null;
+    }
+
+    @Override
+    public HttpIdentityResponse.HttpIdentityResponseBuilder create(IdentityResponse identityResponse) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder responseBuilder =
+                new HttpIdentityResponse.HttpIdentityResponseBuilder();
+        create(responseBuilder, identityResponse);
+        return responseBuilder;
+    }
+
+    @Override
+    public void create(HttpIdentityResponse.HttpIdentityResponseBuilder builder, IdentityResponse identityResponse) {
+
+        SAMLLogoutResponse response = (SAMLLogoutResponse) identityResponse;
+        String samlPostPage = generateSamlPostPage(response.getAcsUrl(), response.getResponse(), response.getRelayState());
+        builder.setBody(samlPostPage);
+        builder.setStatusCode(SC_OK);
+        builder.setRedirectURL(response.getAcsUrl());
+    }
+
+    @Override
+    public HttpIdentityResponse.HttpIdentityResponseBuilder handleException(FrameworkException exception) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder errorResponseBuilder =
+                new HttpIdentityResponse.HttpIdentityResponseBuilder();
+        SAMLIdentityException samlException = (SAMLIdentityException) exception;
+        String samlPostPage = generateSamlPostPage(samlException.getAcsUrl(), samlException.getExceptionMessage(),
+                samlException.getRelayState());
+        errorResponseBuilder.setBody(samlPostPage);
+        errorResponseBuilder.setStatusCode(SC_OK);
+        errorResponseBuilder.setRedirectURL(samlException.getAcsUrl());
+        return errorResponseBuilder;
+    }
+
+    /**
+     * Generate the post page for the logout response.
+     *
+     * @param acUrl       Assertion Service Consumer URL of the Logout Request.
+     * @param samlMessage SAML Logout Response.
+     * @param relayState  RelayState of the Logout Request.
+     * @return Post page.
+     */
+    private String generateSamlPostPage(String acUrl, String samlMessage, String relayState) {
+
+        String postPage = "<html><body><p>You are now redirected back to " + Encode.forHtmlContent(acUrl) +
+                " If the redirection fails, please click the post button.</p><form method='post' action='" +
+                Encode.forHtmlAttribute(acUrl) + "'><p><input type='hidden' name='SAMLResponse' value='"
+                + Encode.forHtmlAttribute(samlMessage) + "'/>";
+
+        if (relayState != null) {
+            postPage = postPage + "<input type='hidden' name='RelayState' value='"
+                    + Encode.forHtmlAttribute(relayState) + "'/>";
+        }
+
+        postPage = postPage + "<button type='submit'>POST</button></p></form><script type='text/javascript'>" +
+                "document.forms[0].submit();</script></body></html>";
+
+        if (log.isDebugEnabled()) {
+            log.debug(postPage);
```

Comment:
better to add a messaging here to indicate that is the post page for the logout response

Reply:
Fixed with PR https://github.com/wso2-extensions/identity-outbound-auth-samlsso/pull/133